### DISCUSSION TITLE
Nuxt without api endpoint

### DISCUSSION THREAD

**Discussion by Atulkumar112** [nuxt/nuxt#25228] [Questions] [answered]
Hi,
I want to use Nuxt without creating api endpoints, means i don't want to use server or server/api folder. i just want to interact my database directly, so is it possible to do that??

- I am using postgre sql database and drizzle ORM

**Comment by TheAlexLichter** [comment]
Sure, you can call any kind of API via useFetch or $fetch. The standalone API can then interact with your DB.

**Comment by curdledSoy** [comment, marked as answer]
i believe there is a couple of modules you can install to assist you here. both by @Hebilicious. @Hebilicious/form-actions-nuxt provides easy implementation of server/form actions similar to next and @Hebilicious/server-block-nuxt which allows you to write these server/form actions and loaders (which works similar to getServerSideProps in next) within your page file.

hope that helps :)

**Reply by Atulkumar112** [reply]
I think it might be an answer which i am looking for. 
Thanks!

**Comment by Atulkumar112** [comment]
Hi @manniL  and @curdledSoy , Thanks for reply.
Actually I am not talking about standalone api. Actually there is a mistake from my side, i had not explain my problem in a right way. sorry for that.
Actually i am new in nuxt. Let me explain my problem again please cooperate me.
In my nuxt app i am using useFetch to get data from my API's which are implemented in server/api directory. so here i want to remove my ./server folder and want to remove my all api endpoints. and want to fetch my data from my pages or components to the database.

So here can i write sql query (using Drizzle ORM) in my component to get data from database. (or my component can interact directly to the database using drizzle ORM).??

**Reply by TheAlexLichter** [reply]
And *why* exactly do you want that?

**Reply by s0h311** [reply]
@Atulkumar112 you should create and use your ORM cliens on the backend side, unless you use something like Supabase that has row level security. This way you avoid unwanted access to you DB. 

Here an example with Prisma:

`/components/ItemsList.vue`

```vue
<template>
  <ul>
    <li v-for="item in items" :key="item.id">
      {{ item.name }}
    </li>
  </ul>
</template>

<script setup>
  const { data: items, error } = await useFetch('/api/items' )
</script>
```

`/server/api/items.get.js`

```JavaScript
export default defineEventHandler((_event) => {
  const prisma = new PrismaClient()

  return prisma.items.findMany()
})
```

**Reply by Hebilicious** [reply]
@Atulkumar112 You can achieve something like that with modules : 

- https://github.com/Hebilicious/server-block-nuxt (experimental)
-  https://github.com/wobsoriano/nuxt-remote-fn by @wobsoriano 

There's probably more options. But that doesn't change the fact that it will still use API endpoints under the hood.

**Reply by japalekhin** [reply]
> And _why_ exactly do you want that?

Am I missing something? I'm new to Nuxt/Drizzle as well, but exposing your db actions to the web via endpoints is dangerous. Let's not even talk about writes.

People can query your data freely without any business logic constraints.

I'm used to PHP/Laravel where I have a service layer that does the actual queries (via Eloquent ORM of course), route handlers and (business logic) api endpoint handlers call the service layer to get data and the service layer queries the database.

I understand that running db queries on a client component is stupid, but if it's still on the SSR run, then that should be way to go. Not get data on a public query endpoint.

My confusion stems from reading tutorials where api endpoints are set up, so now I have `/api/users` (or `/api/users/admins`). I visited that URL and **the data was there!**

**Comment by Atulkumar112** [comment]
> And _why_ exactly do you want that?
I was thinking like php model, so i thought like we could do same here, that's why i asked. Thanks
Anyways nuxt is very cool, am enjoying it. :)

**Reply by TheAlexLichter** [reply]
I strongly suggest to not mix that logic and separate DB access + querying through an API endpoint ☺️
